packages/web/src/interfaces/app-state.ts

```diff
@@ -0,0 +1,7 @@
+import { RouterState } from 'connected-react-router';
+import { AuthState } from './auth-state';
+
+export interface AppState {
+  router: RouterState;
+  auth: AuthState;
+}
```

Comment:
These belong in the store, why did suddenly decide to move them out to interfaces?
Makes no sense to separate the store from the interfaces describing the substores. Is this refactoring make things any better?

Reply:
Well, the idea was to move all the interfaces into one folder to make them easier to track. I believe it to be more convenient but honestly there is no other solid reason.

Reply:
I don't think it makes it easier to track, these interfaces have no meaning outside Redux, and everything Redux-related is co-located in the `store` directory, making it theretically possible to move and reuse together (even though it's not likely to happen). And inside `store` everything is groupped by features.

I agree though that it doesn't really belong in `actions`, so propose to move it to `reducers` instead, right above the initialState using it, and do the same for other substores in the future:

```ts
// store/auth/reducers.ts

export interface AuthState {
  isLoading: boolean;
  isAuthenticated: boolean;
  id: number | null;
  firstName: string | null;
  lastName: string | null;
  email: string | null;
  error: string | null;
  info: string | null;
  socialProfile: SocialProfile | null;
}

export const initialState: AuthState = {
  isAuthenticated: false,
  id: null,
  firstName: null,
  lastName: null,
  email: null,
  isLoading: false,
  error: null,
  info: null
  info: null,
  socialProfile: null
};
```

I believe the main `AppState` should stay in `store/index.ts` though, unless you can propose a better place inside `store`.

Reply:
I don't think anything complex should live in the `index.ts` file. And for sure no file from the `store/` folder should import from it, that does not seem right at all

Reply:
Agree with the `store/index.ts` part.